### PR TITLE
Refactor loadpage: move converter panels to server-side show/hide

### PR DESCRIPTION
**Pull request by swaraj-neu**
## Motivation and context
The loadpage workflow was refactored as a behavior-preserving cleanup to improve maintainability. Client-side `conditionalPanel` logic (and related inline conditions) was replaced with server-side, deterministic visibility predicates that drive show/hide via `shinyjs::toggle()`. The large loadpage server module was split into focused helper components, including migrating converter-related/anomaly-score panels to server-side `renderUI` where applicable. During review, a helper-count documentation mismatch (5 → 6) was corrected.

## Changes
- **Centralized namespaced control identifiers**
  - Added `NAMESPACE_LOADPAGE` in `R/constants.R` to define the UI/server “loadpage” control identifiers used for namespacing and server-driven toggling.

- **Loadpage server split into an orchestrator + helper registrations**
  - Updated `loadpageServer()` in `R/module-loadpage-server.R` to:
    - Remain the orchestrator after the Phase 2 split.
    - Keep only required module state (e.g., editable `condition_metadata`) and `shinyFiles`-based local big-file path wiring for non-web servers.
    - Register helper modules in order:
      - `register_loadpage_preview()`
      - `register_loadpage_visibility_observers()`
      - `register_loadpage_converter_ui()`
      - `register_loadpage_proceed_validation()`
      - `register_loadpage_data_loaders()`
      - `register_loadpage_summary()`
    - Return a reduced interface: `list(input=..., getData=..., getConditionMetadata=...)`.
    - Remove the previous in-file preview/conditional logic and local filename `renderPrint`s.

- **Server-side preview, including DIANN/PTM detection and UI wiring**
  - Added `register_loadpage_preview()` in `R/loadpage-server-preview.R` to:
    - Load a first-100-row preview for supported DIANN and Metamorpheus PTM combinations.
    - Auto-detect whether a DIANN upload is “2+” and synchronize `diann_2plus`, including mismatch warnings.
    - Render PTM modification-ID selection UI from preview content, with support for a custom “__other__” path.

- **Server-side proceed enablement/validation**
  - Added `register_loadpage_proceed_validation()` in `R/loadpage-server-proceed-validation.R` to cascade `proceed1` enable/disable based on:
    - `input$BIO`, `input$DDA_DIA`, and `input$filetype`
    - Required uploaded/reactive inputs per selected pathway (including big-file path length checks where relevant).

- **Event-driven data loading and download/export**
  - Added `register_loadpage_data_loaders()` in `R/loadpage-server-data-loaders.R` to:
    - Provide event-triggered `get_data()` and `get_data_code` derived from `input$calculate`.
    - Implement `output$download_msstats_format` to export either:
      - a single CSV when the result is a `data.frame`, or
      - a ZIP of per-table CSVs while skipping NULL/empty tables and erroring if no non-empty tables exist.
    - Remove `get_summary1`/`get_summary2` from the returned loader object.

- **Post-proceed summary UI + editable condition metadata**
  - Added `register_loadpage_summary()` in `R/loadpage-server-summary.R` to:
    - Create `get_summary1`/`get_summary2` as `eventReactive` computations driven by `proceed1`.
    - Render an editable DT for `condition_metadata` and write edits back into a `reactiveVal`.
    - Initialize template-specific condition metadata from uploaded `data$Condition` (including `protein_turnover` and `chemoproteomics` behaviors, with warning handling on chemoproteomics initialization failure).
    - Show/hide the correct summary table section(s) and advance to the DataProcessing tab on `proceed2`.

- **Converter/anomaly UI moved to server-controlled show/hide + `renderUI`**
  - Added `R/loadpage-server-converter-options-panel.R` providing:
    - A suite of pure `loadpage_show_*` predicate functions controlling visibility for DIANN/LabelFree/OpenSWATH/quantification/annotation/PTM/TMT sections.
    - `register_loadpage_visibility_observers()` that drives panel show/hide via `shinyjs::toggle()` (including BIO-driven toggles for PTM vs non-PTM summary containers after `proceed1`).
    - `register_loadpage_converter_ui()` implementing `renderUI()` outputs for converter fragments (including anomaly-score-related UI via `renderUI` slots) and enabling/disabling/dimming incompatible converter radio choices based on `(BIO, DDA_DIA)`.
    - TMT `which.proteinid` handling unified under a single `renderUI` slot `tmt_options_ui`, with preservation/seed logic for switching between PD ↔ MaxQuant (`loadpage_seed_proteinid()`), tracked with `reactiveVal`.

- **UI refactor: from `conditionalPanel` to server-driven hidden containers**
  - Updated `loadpageUI()` / UI builders in `R/module-loadpage-ui.R` to:
    - Replace `conditionalPanel`-based logic with server-mounted hidden containers (`shinyjs::hidden(...)`) controlled by the server.
    - Migrate major upload/description/panel clusters (including label-free and most PTM/converter uploads) into hidden `div` containers.
    - Render TMT options via `uiOutput(ns(NAMESPACE_LOADPAGE$tmt_options_ui))`.
    - Ensure anomaly UIs use server-driven `renderUI` slots (e.g., Spectronaut anomaly UI), removing reliance on static conditional-panel/JS condition strings.

## Tests
- **Added** `tests/testthat/test-loadpage-server-rendering.R`
  - Truth-table style coverage for loadpage visibility predicate semantics across converter/filetype/LType combinations and key input presence/NULL cases.
  - Phase 1 and Phase 2 assertions for:
    - DIANN LF/options gating, DIANN 2+ gating, q-value/q-cutoff/Mbr/anomaly run-order gating.
    - Sample/label-free selectors and upload-panel visibility across multiple converters and PTM vs non-PTM paths.
    - OpenSWATH m-score cutoff regression requiring the correct ancestor-chain condition.
    - TMT options gating and `NAMESPACE_LOADPAGE` driver/container id assertions.
  - Added tests for:
    - `loadpage_default_proteinid_for_filetype()` defaults.
    - `loadpage_seed_proteinid()` “typed vs default” preservation across PD ↔ MaxQuant switching (including preserved/outgoing edge cases and session reset behavior).
  - Verified post-`proceed1` summary-table predicate behavior for PTM vs non-PTM.

- **Updated** `tests/testthat/test-module-loadpage-ui.R`
  - Adjusted expectations to match the new server-driven hidden-container + `renderUI` architecture.
  - Confirmed:
    - `tmt_options_ui` is provided via `uiOutput(ns("tmt_options_ui"))` and the previous static input is absent.
    - Spectronaut and DIANN anomaly UIs are mounted via server `renderUI` slots and that old static `conditionalPanel`/JS condition strings are no longer present.
    - Large/regular path anomaly gating is expressed via renderUI-time presence of the run-order `fileInput`.

## Coding guidelines violated
- No coding guideline violations were identified in the provided change summary.

### CODE REVIEW

**Comment by swaraj-neu** [comment]
- Fixed the helper-count docs (5 -> 6) which was a real miss from this PR.
- The other five are pre-existing logic that the file split moved out of `module-loadpage-server.R`; none were introduced or changed by this PR, which is intentional since it's a behavior-preserving refactor

File: R/constants.R [hunk 1]
```diff
@@ -61,6 +61,65 @@ CONSTANTS_STATMODEL = list(
   plot_type_qq_plot = "QQPlots" # QQPlots — matches MSstats::groupComparisonQCPlots(type = "QQPlots")
 )
 
+NAMESPACE_LOADPAGE = list(
+  # Cross-module public IDs (read from outside the loadpage module).
```
Comment (tonywu1999) [inline R/constants.R]: Could you remove the comments for the constants, they don't seem to add anything to understanding these constants.

File: R/loadpage-server-data-loaders.R [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+# input-ID renames. Owns:
+#   - 11 single-file wrapper reactives (`get_annot`, `get_annot1/2/3`,
+#     `get_evidence`, `get_evidence2`, `get_global`, `get_proteinGroups`,
+#     `get_proteinGroups2`, `get_FragSummary`, `get_peptideSummary`,
+#     `get_protSummary`, `get_maxq_ptm_sites`)
+#   - the lynchpin `get_data` eventReactive (triggered on `proceed1`)
+#   - the download_msstats_format downloadHandler + its enable/disable
+#     observers
+#   - `get_data_code` (triggered on `calculate`)
+#   - `get_summary1`, `get_summary2` (triggered on `proceed1`)
```
Comment (tonywu1999) [inline R/loadpage-server-data-loaders.R]: Could you get rid of these comments and similar comments?  Seems to be written in a way where you need context from the previous commit

File: R/loadpage-server-data-loaders.R [hunk 3]
```diff
@@ -0,0 +1,177 @@
+# ============================================================================
+# Loadpage — data-loading reactives + download MSstats handler + summaries
```
Comment (tonywu1999) [inline R/loadpage-server-data-loaders.R]: Could you elaborate that the summaries are summary experimental statistics, e.g. number of conditions, number of replicates etc?

File: R/loadpage-server-data-loaders.R [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    getDataCode(input)
+  })
+
+  get_summary1 <- eventReactive(input$proceed1, {
+    getSummary1(input, get_data(), get_annot())
+  })
+
+  get_summary2 <- eventReactive(input$proceed1, {
+    getSummary2(input, get_data())
+  })
```
Comment (tonywu1999) [inline R/loadpage-server-data-loaders.R]: Should this be in loadpage-server-summary.R?

File: R/loadpage-server-summary.R [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    output$template1 <- downloadHandler(
+      filename = function() {
+        paste("extdata/templateevidence", "txt", sep = ".")
+      },
+
+      content = function(file) {
+        file.copy("extdata/templateevidence.txt", file)
+      },
+      contentType = "txt"
+    )
```
Comment (tonywu1999) [inline R/loadpage-server-summary.R]: Could you remind me where we see the template downloads in the loadpage tab?  I don't recall being able to download anything like this.

File: R/loadpage-server-rendering.R [hunk 6]
```diff
@@ -0,0 +1,907 @@
+# ============================================================================
```
Comment (tonywu1999) [inline R/loadpage-server-rendering.R]: Can we rename this file to `loadpage-server-converter-options-panel.R`?

File: R/loadpage-server-summary.R (hunk 5 above)
Comment (swaraj-neu) [reply R/loadpage-server-summary.R]: @tonywu1999 you're right, there's no way to download these in the loadpage tab, they're dead handlers that predate this PR, want me to remove them?

File: R/loadpage-server-data-loaders.R (hunk 3 above)
Comment (swaraj-neu) [reply R/loadpage-server-data-loaders.R]: @tonywu1999 Done. Clarified that these are the experimental-design summary statistics (number of conditions, number of replicates, etc.). Since they're moving to `loadpage-server-summary.R` per your other comment, the clarified description now lives there.

File: R/loadpage-server-summary.R (hunk 5 above)
Comment (tonywu1999) [reply R/loadpage-server-summary.R]: @swaraj-neu Yes, you can remove them.

File: R/loadpage-server-converter-options-panel.R [hunk 7]
```diff
@@ -0,0 +1,856 @@
...
+
+# ----------------------------------------------------------------------------
+# Visibility predicates for the remaining converter and upload panels. Each
+# encodes the full ancestor chain so a nested panel hides when an ancestor does.
+# ----------------------------------------------------------------------------
+
+#' Sample dataset description (parameterized for DDA / DIA / SRM_PRM).
+#' @noRd
+loadpage_show_sample_dataset_description <- function(filetype, label_free_type, mode) {
+  isTRUE(filetype == "sample") && isTRUE(label_free_type == mode)
```
Comment (tonywu1999) [inline R/loadpage-server-converter-options-panel.R]: Confused why `isTRUE(label_free_type == mode)` is needed.

File: R/loadpage-server-converter-options-panel.R [hunk 8]
```diff
@@ -0,0 +1,856 @@
...
+}
+
+#' The LabelFreeType radio (DDA / DIA / SRM_PRM picker) is itself shown only
+#' for the sample-dataset label-free workflow.
+#' @noRd
+loadpage_show_label_free_type_selection <- function(bio, filetype, dda_dia) {
```
Comment (tonywu1999) [inline R/loadpage-server-converter-options-panel.R]: modify the function name so it's clear it's for a sample dataset?

e.g. `loadpage_show_sample_dataset_label_free_type_selector`

File: R/loadpage-server-converter-options-panel.R [hunk 9]
```diff
@@ -0,0 +1,856 @@
...
+loadpage_show_qval_filter <- function(filetype, big_file_diann) {
+  isTRUE(filetype == "sky") ||
+    isTRUE(filetype == "spec") ||
+    (isTRUE(filetype == "diann") && !isTRUE(big_file_diann))
+}
+
+#' @noRd
+loadpage_show_qval_cutoff <- function(q_val) {
+  isTRUE(q_val)
+}
```
Comment (tonywu1999) [inline R/loadpage-server-converter-options-panel.R]: I'd move these functions to a different part of the file since they aren't necessarily DIANN specific (but can involve other converters).

File: R/loadpage-server-converter-options-panel.R [hunk 10]
```diff
@@ -0,0 +1,856 @@
...
+  if (isTRUE(bio == "PTM")) return(FALSE)
+  if (isTRUE(filetype %in% c("sky", "prog", "PD", "open", "spmin", "phil", "meta"))) return(TRUE)
+  if (isTRUE(filetype == "spec")  && !isTRUE(big_file_spec))  return(TRUE)
+  if (isTRUE(filetype == "diann") && !isTRUE(big_file_diann)) return(TRUE)
+  FALSE
+}
+
+#' Pre-formatted MSstats CSV upload — label-free path only.
+#' @noRd
+loadpage_show_msstats_regular_upload <- function(filetype, bio, dda_dia) {
```
Comment (tonywu1999) [inline R/loadpage-server-converter-options-panel.R]: Rename to `loadpage_show_msstats_label_free_upload`

File: R/loadpage-server-converter-options-panel.R [hunk 11]
```diff
@@ -0,0 +1,856 @@
...
+  isTRUE(filetype == "sky") && !isTRUE(bio == "PTM")
+}
+
+#' @noRd
+loadpage_show_ptm_fragpipe_upload <- function(filetype, bio) {
+  isTRUE(filetype == "phil") && isTRUE(bio == "PTM")
+}
+
+#' @noRd
+loadpage_show_maxquant_upload <- function(filetype, bio, dda_dia) {
```
Comment (tonywu1999) [inline R/loadpage-server-converter-options-panel.R:126]: I'm seeing that you have converter-specific functions like `loadpage_show_maxquant_upload` in favor of putting them all in `loadpage_show_standard_quant_upload` because these upload buttons have headers.  Is it overkill to separate the upload button widget from the header widget?  From initial glance and a readability standpoint, it looks really confusing why we're doing all these separate `loadpage_show_maxquant_upload`, `loadpage_show_skyline_upload`, etc. 

This may be the way to go only because I know long term, a lot of differences between templates is purely just headers and text, but the functionality of simply uploading a dataset remains the same.

Comment (swaraj-neu) [reply R/loadpage-server-converter-options-panel.R:126]: @tonywu1999 On the converter-specific upload predicates (maxquant/skyline/etc.): agreed, and it's the right long-term direction. 

The only real difference between most of them is the header text. The upload itself is identical, and the metabolomics template is exactly that case (same upload, different vocabulary).

I'd do it as a follow-up rather than here, since collapsing them into one shared predicate means that predicate has to reproduce the union of all their conditions exactly (the same condition-equivalence risk we've been careful about) and it restructures the UI to pull headers out of the panels, which is more than this behavior-preserving PR. It also folds naturally into the template-adaptable work for metabolomics. 

Happy to scope it as its own task.

File: R/loadpage-server-converter-options-panel.R (hunk 7 above)
Comment (swaraj-neu) [reply R/loadpage-server-converter-options-panel.R]: That clause is what makes each of the three sample-dataset descriptions (DDA / DIA / SRM_PRM) show
only for its matching LabelFreeType, without it all three would show at once. I renamed the param
from `mode` to `label_free_mode` so it reads clearly.

File: R/loadpage-server-converter-options-panel.R [hunk 12]
```diff
@@ -0,0 +1,864 @@
...
+      enable("filetype")
+      disable(selector = "[type=radio][value=prog]")
+      disable(selector = "[type=radio][value=openms]")
+      disable(selector = "[type=radio][value=spec]")
+      disable(selector = "[type=radio][value=open]")
+      disable(selector = "[type=radio][value=ump]")
+      disable(selector = "[type=radio][value=spmin]")
+      disable(selector = "[type=radio][value=diann]")
+      disable(selector = "[type=radio][value=sky]")
+      disable(selector = "[type=radio][value=meta]")
```
Comment (tonywu1999) [inline R/loadpage-server-converter-options-panel.R:908]: Would the code be cleaner if instead of doing this disabling action, we use renderUI to render the list of converter options based on the experiment type here?  And the list of converter options ends up being a widget with input as the list of converter names?  This probably makes sense once we start doing metabolomics templating.

File: R/loadpage-server-summary.R [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+        conditionalPanel(condition = "input['loadpage-BIO'] !== 'PTM'",
+                         h4("Top 6 rows of the dataset"),
+                         div(style = "overflow-x: auto;", tableOutput(ns("summary")))
+        ),
+        conditionalPanel(condition = "input['loadpage-BIO'] == 'PTM'",
+                         h4("Top 6 rows of the PTM dataset"),
+                         div(style = "overflow-x: auto;", tableOutput(ns("summary_ptm"))),
+                         tags$br(),
+                         h4("Top 6 rows of the unmodified protein dataset"),
+                         div(style = "overflow-x: auto;", tableOutput(ns("summary_prot")))
```
Comment (tonywu1999) [inline R/loadpage-server-summary.R]: There's still conditional panels here.  Could you adjust?

File: R/module-loadpage-ui.R [hunk 14]
```diff
@@ -215,61 +219,61 @@ create_file_upload_sections <- function(ns) {
...
       fileInput(ns('msstatsdata'), "", multiple = FALSE, accept = NULL)
-    ),
-    
-    # PTM MSstats format
-    conditionalPanel(
-      condition = "input['loadpage-filetype'] == 'msstats' && (input['loadpage-BIO'] == 'PTM' || (input['loadpage-BIO'] == 'PTM' && input['loadpage-DDA_DIA'] == 'TMT'))",
+    )),
+
+    # PTM MSstats format. (The original JS condition had a redundant TMT
+    # clause that collapsed to `BIO=='PTM'`; the server predicate folds it.)
```
Comment (tonywu1999) [inline R/module-loadpage-ui.R]: You can remove this comment

File: R/module-loadpage-ui.R
Comment (tonywu1999) [inline R/module-loadpage-ui.R]: For both DIANN and Spectronaut, I'm pretty sure you can migrate away from the conditional panel logic here.  I would do renderUI - it doesn't matter that an uploaded file needs to be reuploaded if a user switches from DIANN to Spectronaut or another converter.  The run order file is really small and we can get away with reuploading it.
